### ISSUE TITLE
ncm-spma: userpkgs set to 'no' has no effect

### ISSUE THREAD

**Issue by StephaneGerardVUB** [quattor/configuration-modules-core#1615]
We are using ncm-spma version 21.12.1. Our machines are configured with userpkgs set to 'no', the expected behaviour being that packages installed manually on machines with yum or dnf install commands should be removed by spma. However, it is not the case on EL8 machines. Tracing back the problem in the spma.pm code, it looks like the problem comes from the method packages_to_remove that relies on the command `package-cleanup --leaves --all`. This command returns an empty list on all EL8 machines.

**Comment by wpoely86** [comment]
From https://discussion.fedoraproject.org/t/dnf-list-packages-that-no-other-depends-on/73056/6 I gather this is 'normal'?

We would need https://dnf-plugins-core.readthedocs.io/en/latest/leaves.html to do the equivalent?

**Comment by jrha** [comment]
i.e. #1507?

**Comment by stdweird** [comment]
yes, you need #1507. there reasons why it's not easy to merge are in the PR.
